lib/rpc/ProviderHealthStateDynamoDbRepository.ts

```diff
@@ -0,0 +1,44 @@
+import { ProviderHealthStateRepository } from './ProviderHealthStateRepository'
+import Logger from 'bunyan'
+import { DynamoDB } from 'aws-sdk'
+import { DocumentClient } from 'aws-sdk/clients/dynamodb'
+import { ProviderHealthState } from './ProviderHealthState'
+
+export class ProviderHealthStateDynamoDbRepository implements ProviderHealthStateRepository {
+  private ddbClient: DynamoDB.DocumentClient
+
+  constructor(private dbTableName: string, private log: Logger) {
+    this.ddbClient = new DynamoDB.DocumentClient()
```

Comment:
I would rather we inject a storage interface than newing up the DDB client here, it would increase the inversion of control and the unit testability

Reply:
This whole class `ProviderHealthStateDynamoDbRepository` is exactly made for the purpose you mentioned. You will see in other PR (https://github.com/Uniswap/routing-api/pull/601) that this class is reused to access DB for the online path.

Reply:
not sure how that matters. The current way makes it impossible to unit test, you need an actual DDB